File: Programing_Basics_with_JS/whileLoopExercise/06.Cake.js
function cake(input) {
    let index = 0;
    let cakeWidth = Number(input[index]);
    index++;
    let cakeLenght = Number(input[index]);
    index++;

    let totalPiecesCake = cakeWidth * cakeLenght;
    let command = input[index];
    index++;

    while(command !== 'STOP'){
        let currentPiecesCake = Number(command);
        totalPiecesCake -= currentPiecesCake;

        if (totalPiecesCake < 0) {
            break;
        }

        command = input[index];
        index++;
    }

    if (totalPiecesCake >= 0) {
        console.log(`${totalPiecesCake} pieces are left.`);
    } else {
        console.log(`No more cake left! You need ${Math.abs(totalPiecesCake)} pieces more.`);
    }
}

cake(["10",
"10",
"20",
"20",
"20",
"20",
"21"])

cake(["10",
"2",
"2",
"4",
"6",
"STOP"])

File: Programming_Basics_with_Python/while_loop_exercise/06._cake.py
cake_width = int(input())
cake_length = int(input())
command = input()

cake_size = cake_length * cake_width
while not command == 'STOP':
    cake_pieces = int(command)
    cake_size -= cake_pieces
    if cake_size < 0:
        print(f'No more cake left! You need {abs(cake_size)} pieces more.')
        break

    command = input()

if cake_size >= 0:
    print(f'{cake_size} pieces are left.')